### PR TITLE
[rule-modals] Autocomplete channel

### PR DESCRIPTION
**Pull request by njirap**
Closes-bug: #1257293

### CODE REVIEW

**Comment by bhearsum** [comment]
This code looks OK, but will need a slight update based on my comments on https://github.com/mozilla/balrog/pull/61. You've also got test failures that need fixing - it looks like you need to update some of the tests to take into account the new endpoints you've added, eg: https://github.com/mozilla/balrog-ui/blob/master/spec/controllers/rules_controller_spec.js#L200. Check the "Details" link for more.

File: spec/controllers/releases_controller_spec.js [hunk 1]
```diff
@@ -199,6 +201,8 @@ describe("controller: ReleasesController", function() {
     it("should be possible to open the edit modal", function() {
       this.$httpBackend.expectGET('/api/releases')
       .respond(200, JSON.stringify(sample_releases));
+      this.$httpBackend.expectGET('/api/releases/columns/product')
+      .respond(200, JSON.stringify({product: ['Product1', 'Product2'], count: 2}));
       this.$httpBackend.flush();
```
Comment (njirap) [inline spec/controllers/releases_controller_spec.js]: The existence of these two lines raise a "Unsatisfied request" error while their absence raise a "Unexpected request" error..

File: app/templates/rule_modal.html [hunk 2]
```diff
@@ -12,7 +12,8 @@ <h3 class="modal-title" ng-show="!is_edit && is_duplicate">Duplicate Existing Ru
     </div>
     <div class="form-group" ng-class="{'has-error': errors.product}">
       <label for="id_product">Product</label>
-      <input type="text" class="form-control" id="id_product" ng-model="rule.product">
+      <input type="text" class="form-control" id="id_product" ng-model="rule.product"
+       typeahead="product for product in products | filter:$viewValue | limitTo:16">
       <p class="help-block" ng-show="errors.product">{{ errors.product.join(', ') }}</p>
```
Comment (bhearsum) [inline app/templates/rule_modal.html]: I think you forgot to add the necessary code here to make channel autocomplete. Product works fine in the UI, but channel does not. This is ready to be merged once that's fixed.

Comment (njirap) [inline app/templates/rule_modal.html]: It's done..

**Comment by bhearsum** [comment]
Woohoo, thank you! This will get put into Balrog production soon - hopepfully later this week.

**Pull request merged**